spec/views/pages/conditions/routing_page.html.erb_spec.rb

```diff
@@ -22,13 +22,41 @@
     expect(rendered).to have_css("p.govuk-body", text: t("routing_page.body_text"))
   end
 
-  it "contains a fieldset legend asking a user to select a question page" do
-    expect(rendered).to have_css(".govuk-fieldset__legend", text: t("routing_page.legend_text"))
-    expect(rendered).to have_css("div.govuk-hint", text: t("routing_page.legend_hint_text"))
+  context "with fewer than 10 options" do
+    it "contains a fieldset legend asking a user to select a question page" do
+      expect(rendered).to have_css(".govuk-fieldset__legend", text: t("routing_page.legend_text"))
+      expect(rendered).to have_css("div.govuk-hint", text: t("routing_page.legend_hint_text"))
+    end
+
+    it "has a radio option for each routing pages" do
+      expect(rendered).to have_css(".govuk-radios__item", count: pages.length)
+    end
   end
 
-  it "has a radio option for each routing pages" do
-    expect(rendered).to have_css(".govuk-radios__item", count: pages.length)
+  context "with 10 options" do
+    let(:pages) { build_list :page, 10, :with_selections_settings, form_id: 1 }
+
+    it "contains a fieldset legend asking a user to select a question page" do
+      expect(rendered).to have_css(".govuk-fieldset__legend", text: t("routing_page.legend_text"))
+      expect(rendered).to have_css("div.govuk-hint", text: t("routing_page.legend_hint_text"))
+    end
+
+    it "has a radio option for each routing pages" do
+      expect(rendered).to have_css(".govuk-radios__item", count: pages.length)
+    end
+  end
+
+  context "with more than 10 options" do
+    let(:pages) { build_list :page, 11, :with_selections_settings, form_id: 1 }
+
+    it "contains a fieldset legend asking a user to select a question page" do
+      expect(rendered).to have_css(".govuk-label", text: t("routing_page.legend_text"))
+      expect(rendered).to have_css("div.govuk-hint", text: t("routing_page.legend_hint_text"))
+    end
+
+    it "has a radio option for each routing pages" do
```

Comment:
Is this a typo in the test name? Should it be something like "has a select option for each routing page"?

Reply:
Found my lazy copy-pasting 😬

Reply:
Have updated